api/apiclient.go

```diff
@@ -200,19 +203,12 @@ func Open(info *Info, opts DialOpts) (Connection, error) {
 		httpc := *bakeryClient.Client
 		bakeryClient.Client = &httpc
 	}
-	apiURL, err := url.Parse(dialResult.urlStr)
-	if err != nil {
-		// This should never happen as the url would have failed during dialAPI above.
-		// However the code paths don't allow capture of the url.URL used.
-		return nil, errors.Trace(err)
-	}
-	apiHost := apiURL.Host
 
 	// Technically when there's no CACert, we don't need this
 	// machinery, because we could just use http.DefaultTransport
 	// for everything, but it's easier just to leave it in place.
 	bakeryClient.Client.Transport = &hostSwitchingTransport{
-		primaryHost: apiHost,
+		primaryHost: dialResult.addr,
```

Comment:
// addr is the address used to connect to the API server.

sure sounds like it might have a :PORT in there.
Maybe we could update the docs to say it is either a hostname or an IP address?
Passing 'addr' to 'Host' is always something that feels weird to me, as there is no guarantee an "addr" is a hostname.